include/dts-subscription.sh

```diff
@@ -0,0 +1,341 @@
+  return 1
+}
+
+get_dpp_creds() {
+  echo ""
+  read -p "Enter logs key:                " 'TMP_CLOUDSEND_LOGS_URL'
+  echo ""
+  read -p "Enter firmware download key:   " 'TMP_CLOUDSEND_DOWNLOAD_URL'
+  echo ""
+  read -p "Enter password:                " 'TMP_CLOUDSEND_PASSWORD'
+
+  echo ${TMP_CLOUDSEND_LOGS_URL} > ${SE_credential_file}
+  echo ${TMP_CLOUDSEND_DOWNLOAD_URL} >> ${SE_credential_file}
+  echo ${TMP_CLOUDSEND_PASSWORD} >> ${SE_credential_file}
+
+  print_ok "Dasharo DPP credentials have been saved"
+}
+
+login_to_dpp_server(){
+  # Check if the user is already logged in, log in if not:
+  if [ -z "$(mc alias list | grep ${CLOUDSEND_DOWNLOAD_URL})" ]; then
+    mc alias set $DPP_SERVER_USER_ALIAS $DPP_SERVER_ADDRESS $CLOUDSEND_DOWNLOAD_URL $CLOUDSEND_PASSWORD || return 1
+    if [ $? -ne 0 ]; then
+      print_error "Error while logging in to DPP server!"
+      return 1
+    fi
+  fi
+
+  return 0
+}
+
+subscription_routine(){
+  export CLOUDSEND_LOGS_URL
+  export CLOUDSEND_DOWNLOAD_URL
+  export CLOUDSEND_PASSWORD
+
+  # Each time the main menu is rendered, check for DPP credentials and export
+  # them, if file exists
+  if [ -e "${SE_credential_file}" ]; then
+    CLOUDSEND_LOGS_URL=$(sed -n '1p' < ${SE_credential_file} | tr -d '\n')
+    CLOUDSEND_DOWNLOAD_URL=$(sed -n '2p' < ${SE_credential_file} | tr -d '\n')
+    CLOUDSEND_PASSWORD=$(sed -n '3p' < ${SE_credential_file} | tr -d '\n')
+    export USER_DETAILS="$CLOUDSEND_DOWNLOAD_URL:$CLOUDSEND_PASSWORD"
+    export DPP_IS_LOGGED="true"
+  else
+    CLOUDSEND_LOGS_URL="$BASE_CLOUDSEND_LOGS_URL"
+    CLOUDSEND_PASSWORD="$BASE_CLOUDSEND_PASSWORD"
+    unset CLOUDSEND_DOWNLOAD_URL
+    unset DPP_IS_LOGGED
+    return 1
+  fi
+
+  # Network connection may not be available on boot, do not connect if so:
+  check_network_connection && login_to_dpp_server || return 0
+
+  return 0
+}
+
+check_dasharo_package_env(){
+  [ -d $DPP_PACKAGE_MANAGER_DIR ] || mkdir $DPP_PACKAGE_MANAGER_DIR
+  [ -d $DPP_PACKAGES_SCRIPTS_PATH ] || mkdir $DPP_PACKAGES_SCRIPTS_PATH
+
+  return 0
+}
+
+update_package_list(){
+  check_dasharo_package_env
+
+  mc find --json --name "*.rpm" $DPP_SERVER_USER_ALIAS > $DPP_AVAIL_PACKAGES_LIST
+
+  if [ $? -ne 0 ]; then
+    print_error "Unable to get package list!"
+    return 1
+  fi
+  return 0
+}
+
+download_dpp_package(){
+  local package_name=$1
+
+  # Make sure all paths exist:
+  check_dasharo_package_env
+
+  echo "Downloading package $package_name..."
+
+  # Get package link:
+  local download_link
+  download_link=$(jq -r '.key' "$DPP_AVAIL_PACKAGES_LIST" | grep "$package_name")
+
+  if [ -z "$download_link" ]; then
+    print_error "No package $package_name found!"
+    return 1
+  fi
+
+  # TODO: this will overwrite file with name package_name if its exists, a place
+  # for improvements:
+  local local_path="$DPP_PACKAGE_MANAGER_DIR/$package_name"
+  mc get --quiet "$download_link" "$local_path"
+
+  if [ $? -ne 0 ]; then
```

Comment:
Perhaps `[ $? -ne 0 ] && return 1` if you need only to check error and return

Reply:
Reworked here 6e2ba4b9caaefe8e7cf12a9247479486818efcde.